### PR TITLE
LPS-188159 | Add a testcase to support multitenancy for the headless builder REST APIs

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to support multitenancy for the headless builder REST APIs

Test Plan
CannotSeeAppliationOfOtherInstanceUnderAPIBuilder
When I go to API Builder through Control Panel in the new virtual instance
Then no API application is present

CannotSeeApplicationOfOtherInstanceUnderRestApplications
When I navigate to /o/api in virtual instance
Then /c/{appInDefaultInstance} is not present under REST Applications

Jira ticket:
https://liferay.atlassian.net/browse/LPS-195973

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	property custom.properties = "feature.flag.LPS-167253=true${line.separator}feature.flag.LPS-178642=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginUI();
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:13]: first login here can be change to log through API, we'll need to change bunch of them in the future anyways ;) (FYI https://liferay.atlassian.net/browse/LPS-196890)

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+		task ("Given an published API application with endpoint and schema created in default instance") {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: dont we need these properties to run in default instance? 🤔 
```
property test.liferay.virtual.instance = "false";
		property test.run.type = "single";
```

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	tearDown {
+		HeadlessPortalInstanceAPI.cleanAllPortalInstances();
+
+		ApplicationAPI.deleteAllAPIApplication();
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:50]: if we do this do we still need `HeadlessPortalInstanceAPI.cleanAllPortalInstances();`?

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase (hunk 2 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: for these properties, if we use a virtual instance we don't need them

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase (hunk 3 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:50]: thanks, we don't need `HeadlessPortalInstanceAPI.cleanAllPortalInstances();` any more, will remove it

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
All the failures are about Import/Export functional tests after restricting the entities

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase (hunk 2 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: oh yes, and the step is telling `Given an published API application with endpoint and schema created in default instance`, default instance, right?

Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: right, but in the next we create a virtual instance and log in to it, and all tests use the virtual instance

Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: The thing Meg point is that, we are using **in default instance**, so we actually need to create it in a default instance. And the way of `test.liferay.virtual.instance` is that:
- `test.liferay.virtual.instance` is set to true by default, this way tests will be run on the {server hostname} instance, not the default instance(localhost)
- when we set `test.liferay.virtual.instance = "false"`, tests will be run on the default instance
@Hijky hope this is helpful

Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: got it, thanks, I forgot these cases don't create anything in the virtual instance, will modify it on Monday

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:headless-functional

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures about [ImportExport](https://liferay.atlassian.net/browse/LPS-196791), and the other one failure is about 404 error with API Builder and it also have been fixed

**Review by yanan — APPROVED** [review]

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase (hunk 2 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/APIBuilder-Multitenant/SupportMultitenancyForHeadlessBuilderRESTAPI.testcase:15]: yes, thanks @yanan for clarifying my thought, and sorry for not being clear from the beginning 😅 This was exactly the change I expected, having in mind that the step expresses creating stuff on the default instance :) Thanks for modifying it!

**Review by magjed4289 — APPROVED** [review]
Looks good now, thank you! :)

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**